spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/SimpleRouteLocator.java

```diff
@@ -106,27 +123,49 @@ protected Route getSimpleMatchingRoute(final String path) {
 			log.debug("RequestUtils.isZuulServletRequest()="
 					+ RequestUtils.isZuulServletRequest());
 		}
-
-		String adjustedPath = adjustPath(path);
-
-		ZuulRoute route = getZuulRoute(adjustedPath);
+		
+		String adjustedPath = adjustPath(requestURI);
+		ZuulRoute route = getZuulRoute(adjustedPath, request);
 
 		return getRoute(route, adjustedPath);
 	}
 
-	protected ZuulRoute getZuulRoute(String adjustedPath) {
+	protected ZuulRoute getZuulRoute(String adjustedPath, HttpServletRequest request) {
 		if (!matchesIgnoredPatterns(adjustedPath)) {
-			for (Entry<String, ZuulRoute> entry : this.routes.get().entrySet()) {
+			for (Entry<String, List<ZuulRoute>> entry : this.routes.get()
+					.entrySet()) {
 				String pattern = entry.getKey();
 				log.debug("Matching pattern:" + pattern);
-				if (this.pathMatcher.match(pattern, adjustedPath)) {
-					return entry.getValue();
+				ZuulRoute zuulRoute = this.matchRoute(adjustedPath, entry, request);
+				if (zuulRoute != null){
+					return zuulRoute;
+				}
+			}
+		}
+		return null;
+	}
+	
+	protected ZuulRoute matchRoute(String path, Entry<String, List<ZuulRoute>> entry,
+			HttpServletRequest request) {
+		String method = request.getMethod();
+		String pattern = entry.getKey();
+		List<ZuulRoute> zuulRouteList = entry.getValue();
+		for (ZuulRoute zuulRoute : zuulRouteList) {
```

Comment:
It might be worthwhile to look at https://github.com/spring-cloud/spring-cloud-gateway/blob/2.0.x/spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/GatewayProperties.java to see how I've done similar things. Hard coding matching code seems like a maintenance problem moving forward and combining them should be simpler than nested if's. What if I want to match on method without path? or host and method?